lib/utils.py

```diff
@@ -12,6 +12,23 @@
 
 CURRENT_DIR = os.path.dirname(os.path.abspath(__file__))
 
+def is_outside_symlink(base_dir, symlink_path):
+    try:
+        target = os.readlink(symlink_path)
+        abs_target = os.path.abspath(os.path.join(os.path.dirname(symlink_path), target))
```

Comment:
I think this will still fail if you have a link pointing to a link. I think that is why we used `realpath` before

Reply:
You point out an interesting behaviour and apparently my understanding of .abspath was wrong. I thought it does what .realpath does. Thanks for pointing that out.

I added a test which created the crafted symlink `/tmp/repo/subfolder/../../../etc/passwd` bc then the .startswith comparison makes an abspath look like it is in the repo. but the realpath will resolve to `/etc/passwd`

I also thought about checking if a symlink actually resolves or not at the time of the repo checkout. But tampering afterwards is not really possible as the host os is not touched anymore and the folder is read-only anyway.

Can you give it another look? @ribalba